client/src/templates/Challenges/classic/mobile-layout.tsx

```diff
@@ -45,6 +45,65 @@ class MobileLayout extends Component<MobileLayoutProps, MobileLayoutState> {
     });
   };
 
+  avoidWindowScrolling = (event: Event) => {
+    event.preventDefault();
+  };
+
+  setToolPanelPosition = () => {
+    const toolPanelGroup = (
+      document.getElementsByClassName(
+        'tool-panel-group-mobile'
+      ) as HTMLCollectionOf<HTMLElement>
+    )[0];
+
+    if (navigator.userAgent.match(/iPhone|Android.+Mobile/)) {
+      if (window.innerHeight > (visualViewport?.height as number)) {
+        setTimeout(() => {
+          window.scrollTo(0, 0);
+
+          toolPanelGroup.style.position = 'absolute';
+          toolPanelGroup.style.bottom =
+            String(window.innerHeight - (visualViewport?.height as number)) +
+            'px';
+        }, 200);
+
+        window.addEventListener('touchmove', this.avoidWindowScrolling, {
+          passive: false
+        });
+      } else {
+        document.documentElement.style.height =
+          String(window.innerHeight) + 'px';
+
+        toolPanelGroup.style.position = '';
+        toolPanelGroup.style.bottom = '';
```

Comment:
I didn't seem a usage in these values, can they go or have I missed something?

```suggestion
```

Reply:
This two lines are actually used to solve [original issue](https://github.com/freeCodeCamp/freeCodeCamp/pull/48231#issuecomment-1304376040) of this PR. 
Without these code, the tool panel buttons will not return to normal flow of flex, so they will keep floating and not be anchored at the bottom when address bar is displayed by swiping down.

Reply:
Thank you for linking it, I forgot about that. When I open issue/pr. I will keep it in mind.

Reply:
If opening and closing the top address bar changes the visualViewport.height, can't one logic to handle both scenarios?